### PR TITLE
Update alarm docs to reflect changes in esphome/esphome#5671

### PR DESCRIPTION
**Pull request by hwstar**
## Description:
Update documentation to add on_ready and on_chime callbacks, add sensor_type and zone configuration options, and add get_all_sensors_ready() method.

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<5671>

## Checklist:

  - [X ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/alarm_control_panel/template.rst [hunk 1]
```diff
@@ -36,6 +36,8 @@ Configuration variables:
   - **input** (**Required**, string): The id of the binary sensor component
   - **bypass_armed_home** (*Optional*, boolean): This binary sensor will not trigger the alarm when in ``armed_home`` state.
   - **bypass_armed_night** (*Optional*, boolean): This binary sensor will not trigger the alarm when in ``armed_night`` state.
+  - **sensor_type** (*Optional*, string): Sets the sensor type. One of ``delayed``, ``instant``, or ``interior_follower``. (``delayed`` is default if not specified)
```
Comment (grahambrown11) [inline components/alarm_control_panel/template.rst]: please describe what all these mean

**Review by grahambrown11 — CHANGES_REQUESTED** [review]

File: components/alarm_control_panel/template.rst (hunk 1 above)
Comment (hwstar) [reply components/alarm_control_panel/template.rst]: Docs updated to reflect code changes.

**Comment by hwstar** [comment]
Section on Trigger Modes added to docs.

**Review by jesserockz — DISMISSED** [review]

File: components/alarm_control_panel/index.rst [hunk 2]
```diff
@@ -184,6 +189,39 @@ This trigger is activated when the alarm changes from to disarmed.
         then:
           - logger.log: "Alarm Disarmed!"
 
+.. _alarm_control_panel_on_ready_trigger:
+
+``on_ready`` Trigger
+***********************
+
+This trigger is activated when the logical 'and' of all the alarm sensors change state. This is useful for implementing alarm ready LED's.
```
Comment (kbx81) [inline components/alarm_control_panel/index.rst]:
```suggestion
This trigger is activated when the logical 'and' of all the alarm sensors change state. This is useful for implementing "alarm ready" LEDs.
```

File: components/alarm_control_panel/index.rst [hunk 3]
```diff
@@ -184,6 +189,39 @@ This trigger is activated when the alarm changes from to disarmed.
         then:
           - logger.log: "Alarm Disarmed!"
 
+.. _alarm_control_panel_on_ready_trigger:
+
+``on_ready`` Trigger
+***********************
```
Comment (kbx81) [inline components/alarm_control_panel/index.rst]:
```suggestion
``on_ready`` Trigger
********************
```

File: components/alarm_control_panel/index.rst [hunk 4]
```diff
@@ -184,6 +189,39 @@ This trigger is activated when the alarm changes from to disarmed.
...
+      # ...
+      on_disarmed:
+        then:
+          - lambda: !lambda |-
+              ESP_LOGI("AlarmPanel", "Sensor ready change to: %s", ((id(acp1).get_all_sensors_ready()) ? (const char *) "True" : (const char *) "False"));
+
+.. _alarm_control_panel_on_chime_trigger:
+
+``on_chime`` Trigger
+***********************
```
Comment (kbx81) [inline components/alarm_control_panel/index.rst]:
```suggestion
``on_chime`` Trigger
********************
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Couple little nits, otherwise LGTM

**Review by jesserockz — APPROVED** [review]

**Review by kbx81 — APPROVED** [review]

**Pull request merged**